### PR TITLE
CASMCMS-8329: update working branch template

### PR DESCRIPTION
**Pull request by jesseviola-hpe**
# Description

Adding argo workflow template for `update-working-branch` operation.

[CASMCMS-8329](https://jira-pro.its.hpecorp.net:8443/browse/CASMCMS-8329)

# Checklist Before Merging

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                  value: "{{steps.get-vcs-secrets.outputs.parameters.secret_name}}"
+        - - name: cleanup
+            template: cleanup-template
+            arguments:
+              parameters:
+                - name: vcs_user_credentials_secret_name
+                  value: "{{steps.get-vcs-secrets.outputs.parameters.secret_name}}"
+    ### Templates ###
+    ## get-vcs-secrets-template ##
+    - name: get-vcs-secrets-template
```
Comment (jsollom-hpe) [inline workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml:56]: You've used this in your previous VCS work. Is it possible to extract it into one place for modularity?

**Review by jsollom-hpe — APPROVED** [review]

File: workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml (hunk 1 above)
Comment (jesseviola-hpe) [reply workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml:61]: I agree, I will file a Jira to track this.

File: workflows/iuf/stages.yaml [hunk 2]
```diff
@@ -71,7 +71,7 @@ stages:
     type: product
     operations:
       - name: update-working-branch
```
Comment (atifsyedali) [inline workflows/iuf/stages.yaml:73]: 1. Is the name supposed to be `vcs-update-working-branch`? It is supposed to match the name in the template.
2. The local-path is actually not needed. We need to nuke it from stages.yaml. Just FYI.

File: workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: vcs-update-working-branch
+spec:
+  entrypoint: main
+  arguments:
+    parameters:
+      - name: product_manifest
+      - name: site_params
```
Comment (atifsyedali) [inline workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml:34]: 1. Where do these parameters come from? I thought we were only using `global_params` and `auth_token`. And everything like `product_manifest` and `site_params` will be derived off `global_params`?

2. I don't think the parameters are added under arguments -- they are added as inputs under the step, like the other vcs upload template we edited with Di.

3.

**Review by atifsyedali — CHANGES_REQUESTED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**